sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala

```diff
@@ -184,7 +185,76 @@ private[cosmos] object SparkBridgeImplementationInternal {
       .setIoThreadPriority(config, ioThreadPriority)
   }
 
-  def setUserAgentWithSnapshotInsteadOfBeta() = {
-    HttpConstants.Versions.useSnapshotInsteadOfBeta();
+  def setUserAgentWithSnapshotInsteadOfBeta(): Unit = {
+    HttpConstants.Versions.useSnapshotInsteadOfBeta()
+  }
+
+  def createChangeFeedOffsetFromSpark2
+  (
+    client: CosmosAsyncClient,
+    databaseResourceId: String,
+    containerResourceId: String,
+    tokens: Map[Int, Long]
+  ): String = {
+
+    val databaseName = client
+      .getDatabase(databaseResourceId)
+      .read()
+      .block()
+      .getProperties
+      .getId
+
+    val containerName = client
+      .getDatabase(databaseResourceId)
+      .getContainer(containerResourceId)
+      .read()
+      .block()
+      .getProperties
+      .getId
+
+    val container = client
+      .getDatabase(databaseName)
+      .getContainer(containerName)
+
+    val pkRanges = SparkBridgeInternal
+      .getPartitionKeyRanges(container)
+
+    val continuations = pkRanges
+      .map(pkRange => {
+        val pkRangeId = pkRange.getId.toInt
+
+        val lsn: Long = tokens.applyOrElse(pkRangeId, (_: Int) => {
+          this.logInfo(
+            s"No previous LSN available for PK RangeId $pkRangeId in container $containerName " +
+            s"($containerResourceId) of database $databaseName ($databaseResourceId) - starting from beginning ")
+
+          0L
```

Comment:
do we need to check whether we can find parent pkRangeId in the tokenMap and then using the parent LSN?